aten/src/ATen/native/SoftMax.cpp

```diff
@@ -117,62 +161,62 @@ void host_softmax_backward(
 }
 } // namespace
 
-Tensor softmax_cpu(const Tensor& input_, const int64_t dim_, const bool half_to_float) {
+TORCH_IMPL_FUNC(softmax_cpu_out)
+(const Tensor& input,
+ const int64_t dim,
+ const bool half_to_float,
+ const Tensor& output) {
   TORCH_CHECK(!half_to_float, "softmax with half to float conversion is not supported on CPU");
-  auto input = input_.contiguous();
-  Tensor output = at::native::empty_like(
-      input,
-      c10::nullopt /* dtype */,
-      c10::nullopt /* layout */,
-      c10::nullopt /* device */,
-      c10::nullopt /* pin_memory */,
-      LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  int64_t dim = maybe_wrap_dim(dim_, input.dim());
 
   if (input.numel() == 0) {
-    return output;
+    return;
   }
-  if (input.dim() == 0)
-    input = input.view(1);
+
+  auto input_ = input.contiguous();
```

Comment:
nb: if we ever wanna penny pinch refcounts here, use MaybeOwned, e.g., as in https://github.com/pytorch/pytorch/pull/56115/files this does not block merging this PR

Reply:
a minor style point; personally, I like underscoring the input parameter and then having it un-underscored in the body function; this is mostly based on which variable I'll be referring to a lot and giving it the "better" name. We're very uneven on this everywhere. (also does not block merging the PR)